Windows build errors

Running a basic example, which creates a window and polls events fails using version 0.1 from crates.io on windows:

```rust
   Compiling beryllium v0.1.0
error[E0308]: mismatched types
   --> E:\Rust\registry\src\github.com-1ecc6299db9ec823\beryllium-0.1.0\src\lib.rs:450:9    |
450 |         nn,
    |         ^^ expected enum `core::ffi::c_void`, found enum `winapi::ctypes::c_void`    |
    = note: expected type `core::ptr::NonNull<core::ffi::c_void>`
               found type `core::ptr::NonNull<winapi::ctypes::c_void>`
error[E0308]: mismatched types   --> E:\Rust\registry\src\github.com-1ecc6299db9ec823\beryllium-0.1.0\src\audio.rs:128:58
    |
128 |     let err = unsafe { fermium::SDL_QueueAudio(self.dev, ptr, len) };
    |                                                          ^^^ expected enum `winapi::ctypes::c_void`, found enum `core::ffi::c_void`
    |
    = note: expected type `*const winapi::ctypes::c_void`               found type `*const core::ffi::c_void`
error[E0308]: mismatched types  --> E:\Rust\registry\src\github.com-1ecc6299db9ec823\beryllium-0.1.0\src\cdylib.rs:20:40   |
20 |     unsafe { fermium::SDL_UnloadObject(self.nn.as_ptr()) }
   |                                        ^^^^^^^^^^^^^^^^ expected enum `winapi::ctypes::c_void`, found enum `core::ffi::c_void`
   |   = note: expected type `*mut winapi::ctypes::c_void`
              found type `*mut core::ffi::c_void`

error[E0308]: mismatched types  --> E:\Rust\registry\src\github.com-1ecc6299db9ec823\beryllium-0.1.0\src\cdylib.rs:43:44   |
43 |     NonNull::new(fermium::SDL_LoadFunction(self.nn.as_ptr(), name_ptr))
   |                                            ^^^^^^^^^^^^^^^^ expected enum `winapi::ctypes::c_void`, found enum `core::ffi::c_void`   |
   = note: expected type `*mut winapi::ctypes::c_void`
              found type `*mut core::ffi::c_void`
error[E0308]: mismatched types  --> E:\Rust\registry\src\github.com-1ecc6299db9ec823\beryllium-0.1.0\src\cdylib.rs:43:18
   |
43 |     NonNull::new(fermium::SDL_LoadFunction(self.nn.as_ptr(), name_ptr))
   |                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected enum `core::ffi::c_void`, found enum `winapi::ctypes::c_void`   |
   = note: expected type `*mut core::ffi::c_void`
              found type `*mut winapi::ctypes::c_void`
error[E0308]: mismatched types
   --> E:\Rust\registry\src\github.com-1ecc6299db9ec823\beryllium-0.1.0\src\window.rs:397:20
    |397 |       driver_data: sdl_mode.driverdata,
    |                    ^^^^^^^^^^^^^^^^^^^ expected enum `core::ffi::c_void`, found enum `winapi::ctypes::c_void`
    |
    = note: expected type `*mut core::ffi::c_void`
               found type `*mut winapi::ctypes::c_void`
error[E0308]: mismatched types
   --> E:\Rust\registry\src\github.com-1ecc6299db9ec823\beryllium-0.1.0\src\window.rs:408:19
    |
408 |       driverdata: mode.driver_data,    |                   ^^^^^^^^^^^^^^^^ expected enum `winapi::ctypes::c_void`, found enum `core::ffi::c_void`
    |
    = note: expected type `*mut winapi::ctypes::c_void`
               found type `*mut core::ffi::c_void`

error: aborting due to 7 previous errors

For more information about this error, try `rustc --explain E0308`.
error: Could not compile `beryllium`.
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -15,7 +15,7 @@ categories = ["api-bindings", "multimedia", "game-engines", "no-std"]
 default-target = "x86_64-pc-windows-msvc"
 
 [dependencies]
-fermium = { version = "0.1", default-features = false }
+fermium = { version = "^0.1.2", default-features = false }
 phantom-fields = "^0.1.2"
 raw-window-handle = "0.3"
 

--- a/src/audio.rs
+++ b/src/audio.rs
@@ -123,7 +123,7 @@ impl<'sdl> AudioQueue<'sdl> {
   /// than `u32::MAX` bytes at once.
   pub fn queue_audio(&self, data: &[u8]) -> Result<(), String> {
     assert!(data.len() < core::u32::MAX as usize);
-    let ptr = data.as_ptr() as *const c_void;
+    let ptr = data.as_ptr() as *const fermium::c_void;
     let len = data.len() as u32;
     let err = unsafe { fermium::SDL_QueueAudio(self.dev, ptr, len) };
     if err == 0 {

--- a/src/cdylib.rs
+++ b/src/cdylib.rs
@@ -12,7 +12,7 @@ use super::*;
 #[derive(Debug)]
 #[repr(transparent)]
 pub struct CDyLib<'sdl> {
-  pub(crate) nn: NonNull<c_void>,
+  pub(crate) nn: NonNull<core::ffi::c_void>,
   pub(crate) _marker: PhantomData<&'sdl SDLToken>,
 }
 impl<'sdl> Drop for CDyLib<'sdl> {
@@ -37,7 +37,7 @@ impl<'sdl> CDyLib<'sdl> {
   /// * The returned value _does not_ have a lifetime linking it back to this
   ///   shared library. Making sure that the function pointer is not used after
   ///   this library unloads is up to you.
-  pub unsafe fn find_function(&self, name: &str) -> Option<NonNull<c_void>> {
+  pub unsafe fn find_function(&self, name: &str) -> Option<NonNull<core::ffi::c_void>> {
     let name_null: Vec<u8> = name.bytes().chain(Some(0)).collect();
     let name_ptr: *const c_char = name_null.as_ptr() as *const c_char;
     NonNull::new(fermium::SDL_LoadFunction(self.nn.as_ptr(), name_ptr))

--- a/src/cdylib.rs
+++ b/src/cdylib.rs
@@ -17,7 +17,7 @@ pub struct CDyLib<'sdl> {
 }
 impl<'sdl> Drop for CDyLib<'sdl> {
   fn drop(&mut self) {
-    unsafe { fermium::SDL_UnloadObject(self.nn.as_ptr()) }
+    unsafe { fermium::SDL_UnloadObject(self.nn.as_ptr() as *mut fermium::c_void) }
   }
 }
 impl<'sdl> CDyLib<'sdl> {
@@ -40,6 +40,9 @@ impl<'sdl> CDyLib<'sdl> {
   pub unsafe fn find_function(&self, name: &str) -> Option<NonNull<core::ffi::c_void>> {
     let name_null: Vec<u8> = name.bytes().chain(Some(0)).collect();
     let name_ptr: *const c_char = name_null.as_ptr() as *const c_char;
-    NonNull::new(fermium::SDL_LoadFunction(self.nn.as_ptr(), name_ptr))
+    NonNull::new(
+      fermium::SDL_LoadFunction(self.nn.as_ptr() as *mut fermium::c_void, name_ptr)
+        as *mut core::ffi::c_void,
+    )
   }
 }

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -116,7 +116,6 @@ static A: BerylliumGlobalAlloc = BerylliumGlobalAlloc;
 
 use core::{
   convert::TryFrom,
-  ffi::c_void,
   marker::PhantomData,
   ops::Deref,
   ptr::{null, null_mut, NonNull},
@@ -515,9 +514,9 @@ impl SDLToken {
   }
 
   /// Gets a function pointer to the named OpenGL function
-  pub unsafe fn gl_get_proc_address(&self, name: &str) -> *const c_void {
+  pub unsafe fn gl_get_proc_address(&self, name: &str) -> *const core::ffi::c_void {
     let name_null: Vec<u8> = name.bytes().chain(Some(0)).collect();
-    fermium::SDL_GL_GetProcAddress(name_null.as_ptr() as *const c_char) as *const c_void
+    fermium::SDL_GL_GetProcAddress(name_null.as_ptr() as *const c_char) as *const core::ffi::c_void
   }
 }
 

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -93,9 +93,10 @@
 pub use fermium;
 
 extern crate alloc;
-use alloc::string::String;
-use alloc::string::ToString;
-use alloc::vec::Vec;
+use alloc::{
+  string::{String, ToString},
+  vec::Vec,
+};
 
 #[cfg(feature = "impl_global_alloc")]
 struct BerylliumGlobalAlloc;
@@ -444,7 +445,10 @@ impl SDLToken {
   /// Attempts to load the named dynamic library into the program.
   pub fn load_cdylib<'sdl>(&'sdl self, name: &str) -> Result<CDyLib<'sdl>, String> {
     let name_null: Vec<u8> = name.bytes().chain(Some(0)).collect();
-    match NonNull::new(unsafe { fermium::SDL_LoadObject(name_null.as_ptr() as *const c_char) }) {
+    match NonNull::new(
+      unsafe { fermium::SDL_LoadObject(name_null.as_ptr() as *const c_char) }
+        as *mut core::ffi::c_void,
+    ) {
       Some(nn) => Ok(CDyLib {
         nn,
         _marker: PhantomData,

--- a/src/window.rs
+++ b/src/window.rs
@@ -94,6 +94,7 @@ unsafe impl<'sdl> raw_window_handle::HasRawWindowHandle for Window<'sdl> {
     #[cfg(windows)]
     use raw_window_handle::windows::WindowsHandle;
     use raw_window_handle::RawWindowHandle;
+    use core::ffi::c_void;
     let mut wm_info = fermium::SDL_SysWMinfo::default();
     let b = unsafe { fermium::SDL_GetWindowWMInfo(self.ptr, &mut wm_info) };
     if b == fermium::SDL_TRUE {
@@ -385,7 +386,7 @@ pub struct DisplayMode {
   pub height: i32,
   /// Refresh rate in Hz, or 0 if unspecified.
   pub refresh_rate: i32,
-  driver_data: *mut c_void,
+  driver_data: *mut core::ffi::c_void,
 }
 impl From<fermium::SDL_DisplayMode> for DisplayMode {
   fn from(sdl_mode: fermium::SDL_DisplayMode) -> Self {
@@ -394,7 +395,7 @@ impl From<fermium::SDL_DisplayMode> for DisplayMode {
       width: sdl_mode.w,
       height: sdl_mode.h,
       refresh_rate: sdl_mode.refresh_rate,
-      driver_data: sdl_mode.driverdata,
+      driver_data: sdl_mode.driverdata as *mut core::ffi::c_void,
     }
   }
 }
@@ -405,7 +406,7 @@ impl From<DisplayMode> for fermium::SDL_DisplayMode {
       w: mode.width,
       h: mode.height,
       refresh_rate: mode.refresh_rate,
-      driverdata: mode.driver_data,
+      driverdata: mode.driver_data as *mut fermium::c_void,
     }
   }
 }

--- a/src/window.rs
+++ b/src/window.rs
@@ -81,6 +81,7 @@ impl<'sdl> Drop for Window<'sdl> {
 
 unsafe impl<'sdl> raw_window_handle::HasRawWindowHandle for Window<'sdl> {
   fn raw_window_handle(&self) -> raw_window_handle::RawWindowHandle {
+    use core::ffi::c_void;
     #[cfg(target_os = "macos")]
     use raw_window_handle::macos::MacOSHandle;
     #[cfg(any(
@@ -94,7 +95,6 @@ unsafe impl<'sdl> raw_window_handle::HasRawWindowHandle for Window<'sdl> {
     #[cfg(windows)]
     use raw_window_handle::windows::WindowsHandle;
     use raw_window_handle::RawWindowHandle;
-    use core::ffi::c_void;
     let mut wm_info = fermium::SDL_SysWMinfo::default();
     let b = unsafe { fermium::SDL_GetWindowWMInfo(self.ptr, &mut wm_info) };
     if b == fermium::SDL_TRUE {

